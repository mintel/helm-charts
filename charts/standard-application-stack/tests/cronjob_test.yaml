suite: Test Cronjobs
templates:
  - cronjobs.yaml
release:
  namespace: test-namespace
tests:
  - it: Check CronJob Defaults
    set:
      global.clusterEnv: qa
      cronjobsOnly: true
      cronjobs.jobs:
        - name: daily
          image: ""
          schedule: "0 0 * * *"
          command: [/bin/sh]
          args:
            - "-c"
            - "/app/docker-entrypoint.sh"
    asserts:
      - matchSnapshot: {} # Check for regressions and unexpected changes.
      - isKind:
          of: CronJob
      - equal:
          path: metadata.name
          value: example-app-daily
      - equal:
          path: metadata.namespace
          value: test-namespace
      - hasDocuments:
          count: 1
      - equal:
          path: spec.concurrencyPolicy
          value: Forbid
      - equal:
          path: spec.jobTemplate.spec.template.spec.restartPolicy
          value: Never
      - equal:
          path: spec.schedule
          value: 0 0 * * *
      - equal:
          path: spec.suspend
          value: false
      - equal:
          path: spec.jobTemplate.spec.ttlSecondsAfterFinished
          value: 60
  - it: Check CronJob Default Overrides
    set:
      global.clusterEnv: qa
      cronjobsOnly: true
      cronjobs.defaults:
          concurrencyPolicy: Allow
          suspend: true
          restartPolicy: OnFailure
          ttlSecondsAfterFinished: 120
      cronjobs.jobs:
        - name: daily
          schedule: "0 1 * * *"
          image: ""
          command: [/bin/sh]
          args:
            - "-c"
            - "/app/docker-entrypoint.sh"
    asserts:
      - matchSnapshot: {} # Check for regressions and unexpected changes.
      - isKind:
          of: CronJob
      - equal:
          path: metadata.name
          value: example-app-daily
      - equal:
          path: metadata.namespace
          value: test-namespace
      - hasDocuments:
          count: 1
      - equal:
          path: spec.concurrencyPolicy
          value: Allow
      - equal:
          path: spec.jobTemplate.spec.template.spec.restartPolicy
          value: OnFailure
      - equal:
          path: spec.schedule
          value: 0 1 * * *
      - equal:
          path: spec.suspend
          value: true
      - equal:
          path: spec.jobTemplate.spec.ttlSecondsAfterFinished
          value: 120
  - it: Check CronJob Job Overrides
    set:
      global.clusterEnv: qa
      cronjobsOnly: true
      cronjobs.jobs:
        - name: daily
          concurrencyPolicy: Allow
          suspend: true
          restartPolicy: OnFailure
          schedule: "0 1 * * *"
          ttlSecondsAfterFinished: 120
          image: ""
          command: [/bin/sh]
          args:
            - "-c"
            - "/app/docker-entrypoint.sh"
    asserts:
      - matchSnapshot: {} # Check for regressions and unexpected changes.
      - isKind:
          of: CronJob
      - equal:
          path: metadata.name
          value: example-app-daily
      - equal:
          path: metadata.namespace
          value: test-namespace
      - hasDocuments:
          count: 1
      - equal:
          path: spec.concurrencyPolicy
          value: Allow
      - equal:
          path: spec.jobTemplate.spec.template.spec.restartPolicy
          value: OnFailure
      - equal:
          path: spec.schedule
          value: 0 1 * * *
      - equal:
          path: spec.suspend
          value: true
      - equal:
          path: spec.jobTemplate.spec.ttlSecondsAfterFinished
          value: 120
  - it: Check CronJob ttlSecondsAfterFinished zero value
    set:
      global.clusterEnv: qa
      cronjobsOnly: true
      cronjobs.defaults:
          ttlSecondsAfterFinished: 0
      cronjobs.jobs:
        - name: daily
          schedule: "0 1 * * *"
          image: ""
          command: [/bin/sh]
          args:
            - "-c"
            - "/app/docker-entrypoint.sh"
    asserts:
      - matchSnapshot: {} # Check for regressions and unexpected changes.
      - isKind:
          of: CronJob
      - equal:
          path: metadata.name
          value: example-app-daily
      - equal:
          path: metadata.namespace
          value: test-namespace
      - hasDocuments:
          count: 1
      - isNull:
          path: spec.jobTemplate.spec.ttlSecondsAfterFinished
