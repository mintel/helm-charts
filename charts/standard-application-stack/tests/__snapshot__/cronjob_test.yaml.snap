Check CronJob Default Overrides:
  1: |
    apiVersion: batch/v1
    kind: CronJob
    metadata:
      annotations:
        helm.sh/chart: standard-application-stack-5.4.0
      labels:
        app.kubernetes.io/component: daily
        app.kubernetes.io/managed-by: Helm
        app.kubernetes.io/name: example-app-daily
        app.mintel.com/env: qa
        app.mintel.com/region: ${CLUSTER_REGION}
        name: example-app-daily
      name: example-app-daily
      namespace: test-namespace
    spec:
      concurrencyPolicy: Allow
      jobTemplate:
        spec:
          template:
            metadata:
              labels:
                app.kubernetes.io/component: daily
                app.kubernetes.io/managed-by: Helm
                app.kubernetes.io/name: example-app-daily
                app.mintel.com/env: qa
                app.mintel.com/region: ${CLUSTER_REGION}
                name: example-app-daily
            spec:
              containers:
                - args:
                    - -c
                    - /app/docker-entrypoint.sh
                  command:
                    - /bin/sh
                  env:
                    - name: KUBERNETES_POD_NAMESPACE
                      valueFrom:
                        fieldRef:
                          fieldPath: metadata.namespace
                    - name: APP_ENVIRONMENT
                      value: qa
                    - name: RUNTIME_ENVIRONMENT
                      value: kubernetes
                  envFrom: null
                  image: registry.gitlab.com/test:v0.1.0
                  imagePullPolicy: IfNotPresent
                  name: main
              imagePullSecrets:
                - name: image-pull-gitlab
                - name: image-pull-docker-hub
              restartPolicy: OnFailure
              serviceAccountName: example-app
          ttlSecondsAfterFinished: 120
      schedule: 0 1 * * *
      suspend: true
Check CronJob Defaults:
  1: |
    apiVersion: batch/v1
    kind: CronJob
    metadata:
      annotations:
        helm.sh/chart: standard-application-stack-5.4.0
      labels:
        app.kubernetes.io/component: daily
        app.kubernetes.io/managed-by: Helm
        app.kubernetes.io/name: example-app-daily
        app.mintel.com/env: qa
        app.mintel.com/region: ${CLUSTER_REGION}
        name: example-app-daily
      name: example-app-daily
      namespace: test-namespace
    spec:
      concurrencyPolicy: Forbid
      jobTemplate:
        spec:
          template:
            metadata:
              labels:
                app.kubernetes.io/component: daily
                app.kubernetes.io/managed-by: Helm
                app.kubernetes.io/name: example-app-daily
                app.mintel.com/env: qa
                app.mintel.com/region: ${CLUSTER_REGION}
                name: example-app-daily
            spec:
              containers:
                - args:
                    - -c
                    - /app/docker-entrypoint.sh
                  command:
                    - /bin/sh
                  env:
                    - name: KUBERNETES_POD_NAMESPACE
                      valueFrom:
                        fieldRef:
                          fieldPath: metadata.namespace
                    - name: APP_ENVIRONMENT
                      value: qa
                    - name: RUNTIME_ENVIRONMENT
                      value: kubernetes
                  envFrom: null
                  image: registry.gitlab.com/test:v0.1.0
                  imagePullPolicy: IfNotPresent
                  name: main
              imagePullSecrets:
                - name: image-pull-gitlab
                - name: image-pull-docker-hub
              restartPolicy: Never
              serviceAccountName: example-app
          ttlSecondsAfterFinished: 60
      schedule: 0 0 * * *
      suspend: false
Check CronJob Job Overrides:
  1: |
    apiVersion: batch/v1
    kind: CronJob
    metadata:
      annotations:
        helm.sh/chart: standard-application-stack-5.4.0
      labels:
        app.kubernetes.io/component: daily
        app.kubernetes.io/managed-by: Helm
        app.kubernetes.io/name: example-app-daily
        app.mintel.com/env: qa
        app.mintel.com/region: ${CLUSTER_REGION}
        name: example-app-daily
      name: example-app-daily
      namespace: test-namespace
    spec:
      concurrencyPolicy: Allow
      jobTemplate:
        spec:
          template:
            metadata:
              labels:
                app.kubernetes.io/component: daily
                app.kubernetes.io/managed-by: Helm
                app.kubernetes.io/name: example-app-daily
                app.mintel.com/env: qa
                app.mintel.com/region: ${CLUSTER_REGION}
                name: example-app-daily
            spec:
              containers:
                - args:
                    - -c
                    - /app/docker-entrypoint.sh
                  command:
                    - /bin/sh
                  env:
                    - name: KUBERNETES_POD_NAMESPACE
                      valueFrom:
                        fieldRef:
                          fieldPath: metadata.namespace
                    - name: APP_ENVIRONMENT
                      value: qa
                    - name: RUNTIME_ENVIRONMENT
                      value: kubernetes
                  envFrom: null
                  image: registry.gitlab.com/test:v0.1.0
                  imagePullPolicy: IfNotPresent
                  name: main
              imagePullSecrets:
                - name: image-pull-gitlab
                - name: image-pull-docker-hub
              restartPolicy: OnFailure
              serviceAccountName: example-app
          ttlSecondsAfterFinished: 120
      schedule: 0 1 * * *
      suspend: true
Check CronJob ttlSecondsAfterFinished zero value:
  1: |
    apiVersion: batch/v1
    kind: CronJob
    metadata:
      annotations:
        helm.sh/chart: standard-application-stack-5.4.0
      labels:
        app.kubernetes.io/component: daily
        app.kubernetes.io/managed-by: Helm
        app.kubernetes.io/name: example-app-daily
        app.mintel.com/env: qa
        app.mintel.com/region: ${CLUSTER_REGION}
        name: example-app-daily
      name: example-app-daily
      namespace: test-namespace
    spec:
      concurrencyPolicy: Forbid
      jobTemplate:
        spec:
          template:
            metadata:
              labels:
                app.kubernetes.io/component: daily
                app.kubernetes.io/managed-by: Helm
                app.kubernetes.io/name: example-app-daily
                app.mintel.com/env: qa
                app.mintel.com/region: ${CLUSTER_REGION}
                name: example-app-daily
            spec:
              containers:
                - args:
                    - -c
                    - /app/docker-entrypoint.sh
                  command:
                    - /bin/sh
                  env:
                    - name: KUBERNETES_POD_NAMESPACE
                      valueFrom:
                        fieldRef:
                          fieldPath: metadata.namespace
                    - name: APP_ENVIRONMENT
                      value: qa
                    - name: RUNTIME_ENVIRONMENT
                      value: kubernetes
                  envFrom: null
                  image: registry.gitlab.com/test:v0.1.0
                  imagePullPolicy: IfNotPresent
                  name: main
              imagePullSecrets:
                - name: image-pull-gitlab
                - name: image-pull-docker-hub
              restartPolicy: Never
              serviceAccountName: example-app
      schedule: 0 1 * * *
      suspend: false
